implementations/rust/ockam/ockam_core/src/access_control.rs

```diff
@@ -26,15 +26,15 @@ use crate::{LocalMessage, Result};
 #[async_trait]
 pub trait AccessControl: Send + Sync + 'static {
     /// Return true if the message is allowed to pass, and false if not.
-    async fn is_authorized(&mut self, local_msg: &LocalMessage) -> Result<bool>;
+    async fn is_authorized(&self, local_msg: &LocalMessage) -> Result<bool>;
```

Comment:
This is debatable of course. Currently no impl requires unique access, but if the design is intended to do so, we may instead want to add `#[allow(clippy::wrong-self-convention)]`.

Reply:
I think there used to be the perception in ockam's code that we should prefer `&mut self`, but I'm strongly in favor of not doing so, and places where we don't need it should remove it. Clippy is right here.

Anybody who needs mutability in this trait can hold onto a Mutex or something.